packages/kbn-apm-synthtrace/src/lib/logs/logs_synthtrace_es_client.ts

```diff
@@ -48,6 +48,33 @@ export class LogsSynthtraceEsClient extends SynthtraceEsClient<LogDocument> {
     }
   }
 
+  async createComponentTemplate(name: string, mappings: MappingTypeMapping) {
+    const isTemplateExisting = await this.client.cluster.existsComponentTemplate({ name });
+
+    if (isTemplateExisting) return this.logger.info(`Component template already exists: ${name}`);
+
+    try {
+      await this.client.cluster.putComponentTemplate({
+        name,
+        template: {
+          mappings,
+        },
+      });
+      this.logger.info(`Component template successfully created: ${name}`);
+    } catch (err) {
+      this.logger.error(`Component template creation failed: ${name} - ${err.message}`);
+    }
+  }
+
+  async deleteComponentTemplate(name: string) {
+    try {
+      await this.client.cluster.deleteComponentTemplate({ name });
+      this.logger.info(`Component template successfully deleted: ${name}`);
+    } catch (err) {
+      this.logger.error(`Component template deletion failed: ${name} - ${err.message}`);
+    }
```

Comment:
Is there any potential issues for the following tests if it component template was not deleted?

Reply:
If component template is not deleted, then there is no issue as a CT is hooked to an Index template and i am deleting the Index Template.

if the index template fails to delete, it is possible it might impact other tests, which use exactly the same data stream name.

To make it full proof. i am using custom names for the data stream and hence Index template and CT both are exclusive to this DS